src/main/java/org/truffleruby/core/encoding/TranscodingManager.java

```diff
@@ -82,16 +62,16 @@ public Transcoder getTranscoder() {
                 final String sourceName = new String(e.getSource()).toUpperCase();
                 final String destinationName = new String(e.getDestination()).toUpperCase();
 
-                final TranscoderReference transcoder;
+                final Transcoder transcoder;
                 if (TruffleOptions.AOT) {
                     // Load the classes eagerly
-                    transcoder = new TranscoderReference(e.getTranscoder(), null);
+                    transcoder = e.getTranscoder();
                 } else {
-                    transcoder = new TranscoderReference(null, e);
+                    transcoder = null;
```

Comment:
Isn't this going to set the the entry value to `null` on JVM?

Does getting the Transcoder instance implies the heavy class-loading + data files parsing?
If not it sounds fine to e.getTranscoder() in both cases (but better check with `jt metrics time`).